Add keyCode mapping for browsers that don't emit a `event.key` property

simontsang opened the pull request:
Older browsers don't have a `KeyboardEvent.key` property, and so requires a fallback to the legacy `KeyboardEvent .keyCode`

This pulls in the changes from https://github.com/mgroh/lrud/tree/webos-keycodes but modifies the behaviour slightly to use the `key` property if it exists.

This also maps the `Backspace` key to "back" actions, as used on certain browser environments (e.g. PlayStation 4)

mgroh committed:
Use 'keycode' instead of 'key' to map KeyboardEvents to bindings

On some platforms (e.g. webOS 3) a KeyboardEvent only contains a keycode and no key

mgroh committed:
Use 'keycode' instead of 'key' to map KeyboardEvents to bindings

Map webos back keycode

mgroh committed:
Map Tizen back keycode

simontsang committed:
Add binding to "back" for backspace key code

simontsang committed:
Use `.keyCode` only as a fallback to `.key`

simontsang committed:
Bind backspace key to "back" action

simontsang committed:
Fallback to keyCode if no key mapping found

mgroh commented on src/lrud-input/code-to-binding-map.ts:
```diff
@@ -0,0 +1,12 @@
+// This maps a key code, returned from an event, to a handler name.
+// TODO: add configuration of custom key map
```
That TODO is what I mentioned, allowing the whole keymap to be configured. As discussed this might not be necessary, so I think we can delete the TODO here

mgroh reviewed (approved):
💯

simontsang committed:
docs: Remove TODO from code-to-binding-map.ts

As mentioned in https://github.com/mubi/lrud/pull/2#pullrequestreview-797538940,
the TODO is perhaps no longer necessary
